### PR TITLE
Allow happy access to children in SubRecords

### PR DESCRIPTION
**Pull request by zakird**

### CODE REVIEW

**Comment by zakird** [comment]
This should make @cdzombak's life happier in the future.

File: zschema/compounds.py [hunk 1]
```diff
@@ -133,6 +133,18 @@ def __init__(self, definition=_NO_ARG, extends=_NO_ARG,
             for k, v in sorted(self.definition.iteritems()):
                 _is_valid_object(k, v)
 
+    def __getitem__(self, key):
```
[the remark this answers is not in the document]
Comment (justinbastress) [reply zschema/compounds.py]: Ah, I think I see it...the key in question is "tls" (https://github.com/zmap/zschema/blob/master/zschema/tests.py#L505), which has `definition == {}`, which evaluates to `False`...and `__len__` is making this bubble out to the `SubRecord` itself:

```python
class NoLen(object):
  pass

class HasLen(object):
  def __init__(self, x):
    self.x = x

  def __len__(self):
    return self.x


>>> a=HasLen(0)
>>> b=HasLen(1)
>>> a and "yes" or "no"
'no'
>>> b and "yes" or "no"
'yes'

>>> c=NoLen()
>>> c and "yes" or "no"
'yes'
```

**Comment by zakird** [comment]
Nice catch. I've pulled `__len__`.

**Comment by andrewsardone** [comment]
>> Apparently `__len__() == 0` makes an object evaluate to `False`
>
> Nice catch. I've pulled `__len__`.

Cool to approve and merge, @justinbastress? ✨

**Comment by zakird** [comment]
Justin, can you confirm you're cool with this?

**Pull request merged**